packages/desktop-client/src/components/App.tsx

```diff
@@ -140,9 +140,17 @@ function AppWrapper() {
       }
     }
 
+    function onVisibilityChange() {
+      sync();
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏the sync operation is not always super quick. If the user switches back and forth between the windows - this will dispatch multiple sync events. Would it be possible to avoid this?

I'm thinking either wait for the active sync even to finish before allowing a new one. Or if that's very complex, then maybe just a delay between sync-s?

WDYT?

Reply:
Good idea! Implemented in https://github.com/actualbudget/actual/pull/1549/commits/765674878abe824974f9c21f910991f8b0c6ae8c.